extensions/amp-a4a/0.1/name-frame-renderer.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Renderer} from './amp-ad-type-defs';
+import {createElementWithAttributes} from '../../../src/dom';
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getContextMetadata} from '../../../src/iframe-attributes';
+import {getDefaultBootstrapBaseUrl} from '../../../src/3p-frame';
+import {utf8Decode} from '../../../src/utils/bytes';
+
+/**
+ * Render a non-AMP creative into a NameFrame.
+ */
+export class NameFrameRenderer extends Renderer {
+  /** @override */
+  render(context, element, unusedCreativeData) {
+    const crossDomainData = context.crossDomainData;
```

Comment:
shall we pass creative data as the `creativeData` arg, instead of `context`

Reply:
`creativeData` is for any information that the validator needs to pass to the renderer, whereas the context is for the AmpAdNetworkBase class to pass information through validation/rendering. For the NameFrameRenderer, there is no validation, so we don't use `creativeData`.

Reply:
it will be good to keep the same path for all type of ads. we can take it as a dummy "always pass" validator.

i want to avoid having everything in `context`. all common/required data should be explicitly passed in as other params.

Reply:
Well the flow would be something like:

CryptoValidator invoked -> CryptoValidator returns non-AMP response -> NameFrameRenderer is invoked. 

I don't think it would make much sense for the validator in that case to pass nameframe specific data.

Reply:
I don't quite understand how the current code work. Is there other changes you didn't include in this PR? For example, where is the code setting the creative data to context?

I agree that validator should not do anything special to nameframe, it can just throw the non-AMP response to the nameframe, all nameframe specific logic  should happen there.

Reply:
that is not ideal, because nameframe specific logic would appear in that subclass.

can we just pass the ad server response directly to the renderer after validation?
is srcUrl the ad request URL ? I'm fine to have that in context object.

Reply:
Okay. I think we can pass this data through `creativeData` as you suggest. If anything, we can iron out the kinks once we start using the renderer.